lib/paper_trail/has_paper_trail.rb

```diff
@@ -127,6 +127,18 @@ def paper_trail
         ::PaperTrail::RecordTrail.new(self)
       end
 
+      def dependent_versions(class_name, foreign_key)
+        children = []
+        versions = PaperTrail::Version.where(item_type: class_name)
```

Comment:
I am not sure what you mean by polymorphous associations because the foreign_key argument can be any method or attribute of the class.